src/NServiceBus.Core.Tests/Licensing/LicenseManagerTests.cs

```diff
@@ -0,0 +1,279 @@
+        {
+            var licenseManager = new LicenseManager();
+            licenseManager.todayUtc = () => DateTime.ParseExact(currentDate, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal);
+            var logger = new TestableLogger();
+            var license = new License
+            {
+                UpgradeProtectionExpiration = DateTime.ParseExact(expirationDate, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal),
+                LicenseType = "not-trial"
+            };
+
+            licenseManager.LogExpiringLicenseWarning(license, logger);
+
+            Assert.AreEqual(1, logger.Logs.Count);
+            Assert.AreEqual("Please extend your upgrade protection so that we can continue to provide you with support and new versions of the Particular Service Platform.", logger.Logs.Single().Item1);
+            Assert.AreEqual(LogLevel.Warn, logger.Logs.Single().Item2);
+        }
+
+        [TestCase("2012-12-12", "2012-12-16")]
+        [TestCase("2012-12-12", "2012-12-17")]
+        public void ShouldNotWarnAboutExpiringUpgradeProtectionWhenMoreThanThreeDaysAway(string currentDate, string expirationDate)
+        {
+            var licenseManager = new LicenseManager();
+            licenseManager.todayUtc = () => DateTime.ParseExact(currentDate, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal);
+            var logger = new TestableLogger();
+            var license = new License
+            {
+                UpgradeProtectionExpiration = DateTime.ParseExact(expirationDate, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal),
+                LicenseType = "not-trial"
+            };
+
+            licenseManager.LogExpiringLicenseWarning(license, logger);
+
+            Assert.AreEqual(0, logger.Logs.Count);
+        }
+
+        [Test]
+        public void ShouldLogErrorAboutExpiredUpgradeProtection()
+        {
+            var licenseManager = new LicenseManager();
+            licenseManager.todayUtc = () => new DateTime(2012, 12, 12);
+            var logger = new TestableLogger();
+            var license = new License
+            {
+                UpgradeProtectionExpiration = new DateTime(2010, 10, 10),
+                LicenseType = "not-trial"
+            };
+
+            licenseManager.LogExpiredLicenseError(license, logger);
+
+            Assert.AreEqual(1, logger.Logs.Count);
+            Assert.AreEqual("Please extend your upgrade protection so that we can continue to provide you with support and new versions of the Particular Service Platform.", logger.Logs.Single().Item1);
+            Assert.AreEqual(LogLevel.Error, logger.Logs.Single().Item2);
+        }
+
+        class TestableLogger : ILog
```

Comment:
Was https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Testing.Fakes/DefaultTestingLoggerFactory.cs#L7 not suitable for this? (or was it only intended for our end users)

Reply:
the testingloggerfactory is more complicated to use than this really trivial implementation.

Reply:
Ok, happy with that